scripts/azkaban_job_whitelisting_blacklisting.py

```diff
@@ -0,0 +1,169 @@
+	cursor.execute(get_all_tuning_auto_apply_azkaban_rules_query)
+	row=cursor.fetchone()
+	while row is not None: 
+		rule_type = row["rule_type"]
+		project_name = row["project_name"]
+		flow_name_expr = row["flow_name_expr"]
+		job_name_expr = row["job_name_expr"]
+		job_type = row["job_type"]
+		rule=Rule(rule_type, project_name, flow_name_expr, job_name_expr, job_type)
+		if(ruleList.has_key(project_name)):
+			projectRuleList=ruleList[project_name]
+		else : 
+			projectRuleList=[]
+			ruleList[project_name] = projectRuleList
+		projectRuleList.append(rule)
+		row = cursor.fetchone()
+	return ruleList
+
+def update_auto_apply_flag(records_to_update): 
+    try:
+    	conn.autocommit = True 
+    	cursor = conn.cursor()
+    	cursor.executemany(update_auto_apply_flag_query, records_to_update)
+
+    except Exception as e:
+    	conn.rollback()
+    	logger.error(e)
+
+    finally:
+    	cursor.close()
+
+def update_auto_apply_flag_for_jobs(jobs):
+	records_to_update = []
+	for job in jobs : 
+		if job.auto_apply!=job.new_auto_appply:
+			if len(records_to_update) >= BATCH_SIZE : 
+				update_auto_apply_flag(records_to_update)
+				records_to_update = []
+			batch_element = (job.new_auto_appply, job.job_definition_id)
+			updated_job_definition_ids[job.new_auto_appply].append(job.job_definition_id)
+			records_to_update.append(batch_element)
+
+	if len(records_to_update)>0 : 
+		update_auto_apply_flag(records_to_update)
+
+class Job:
+	def __init__(self, _job_definition_id, _job_def_id, _auto_apply, _job_type):
+		self.job_definition_id = _job_definition_id
+		self.job_def_id = _job_def_id
+		self.auto_apply = _auto_apply
+		self.job_type = _job_type
+
+	def set_new_auto_apply_value(self, auto_apply):
+		self.new_auto_appply=auto_apply
+
+	def apply_rules(self, rules):
+		auto_apply=False
+		for rule in rules :
+			if self.job_type == rule.job_type : 
+				pattern = self.get_flow_job_expr(rule)
+				if re.match(pattern, self.job_def_id) :
+					if rule.rule_type=="WHITELIST" : 
+						auto_apply=True 
+					elif rule.rule_type=="BLACKLIST" : 
+						auto_apply=False
+						break
+		return auto_apply
+
+	def get_flow_job_expr(self, rule):
+		pattern=".*flow=" + rule.flow_name_expr + "&job=" + rule.job_name_expr + "$" 
+		return pattern
+
+class Rule:
+	def __init__(self, _rule_type, _project_name, _flow_name_expr, _job_name_expr, _job_type):
+		self.rule_type = _rule_type
+		self.project_name = _project_name
+		self.flow_name_expr = _flow_name_expr
+		self.job_name_expr = _job_name_expr
+		self.job_type = _job_type
+
+	def print_rule(self):
+		logger.info("Rule Type: " + self.rule_type + ", Project Name: " + self.project_name + ", Flow Name Expr: " + self.flow_name_expr + ", Job Name Expr: " + self.job_name_expr + ", Job Type: " + self.job_type)
+
+
+def print_rules(rules):
+	for rule in rules:
+		rule.print_rule()
+
+conn=get_mysql_connection()
+logger.info("mysql connection established")
+
+rulesMap = get_all_tuning_auto_apply_azkaban_rules()
```

Comment:
How have u unit tested the code ?

Reply:
Couldn't do unit test cases as I couldn't get H2 connection created in Python script.